### PR TITLE
heapsort implemented

### PR DESCRIPTION
**Pull request by SrishtiSengupta**
references #4 and #12 

-Team l1ghtsab3r

### CODE REVIEW

File: pydsa/heap_sort.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+"""
+    Heap Sort
```
Comment (malayaleecoder) [inline pydsa/heap_sort.py]: I think you don't have to copy the same comment from below. According to the previous sorting algorithms implemented only the name and complexity would do here. @aktech What's your view on this?

**Comment by CuriousLearner** [comment]
@SrishtiSengupta You need to resolve conflicts first.

File: pydsa/heap_sort.py (hunk 1 above)
Comment (aktech) [inline pydsa/heap_sort.py]: +1

**Comment by goelakash** [comment]
@SrishtiSengupta still on this?

**Comment by SrishtiSengupta** [comment]
@aktech @goelakash The merge conflicts have been resolved.
Sorry for the delay, I was a little caught up with end semester exams & project demos.

File: pydsa/tests/test_heap_sort.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from pydsa.heap_sort import heap_sort
+from random import randint
+
+def test_heap_sort():
+    a = [randint(1, 10) for i in range(10)]
+    b = sorted(a)
+    heap_sort(a)
+    assert a == b
```
Comment (goelakash) [inline pydsa/tests/test_heap_sort.py]: Please add a newline character here.

**Comment by goelakash** [comment]
@SrishtiSengupta Please squash the commits when you're done. We'll merge it. Thanks for the work BTW!

File: pydsa/heap_sort.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    if left < end and list_item[i] < list_item[left]:
+        max = left
+    if right < end and list_item[max] < list_item[right]:
+        max = right
+    if max != i:
+        swap(list_item, i, max)
+        heapify(list_item, end, max)
+
+def swap(list_item, i, j):
+    list_item[i], list_item[j] = list_item[j], list_item[i]
```
Comment (CuriousLearner) [inline pydsa/heap_sort.py]: Add new line at the end of file.

**Comment by goelakash** [comment]
@SrishtiSengupta You can set up your editor to add new-line at end-of-file on file save. For example, to do this on sublime text, check this - http://docs.sublimetext.info/en/latest/reference/settings.html#automatic-behavior

**Pull request merged**